### ISSUE TITLE
Version

### ISSUE THREAD

**Issue by impredicative** [EricAlcaide/SimpleNet-Keras#2]
Which version of the paper or the Caffe code is this an implementation of? It'll help to declare this explicitly in the readme. Also, it'd be great to bring it up to date with the current latest version of the paper, i.e. v5.

**Comment by hypnopump** [comment]
I think the version of the paper doesn't affect the main architecture of the model. In the recent versions, the paper has introduced slim/lightweight architectures that, however, show same results and number of parameters as WRN 10-2, so I don't know what's really new there.

The model architecture has been extracted from the Caffe .prototxt file from the original repository: 
https://github.com/Coderx7/SimpleNet/blob/master/SimpNet_V1/Models/DropoutAll/train_val.prototxt however, the data augmentation scheme is not the same. 

I'll update the readme with results of the accuracy achieved with the model, provide pretrained models and weights (.h5 files) and I'll also try to modify the structure of the file in order to be able to call it from a notebook or another file.

**Comment by Coderx7** [comment]
@EricAlcaide I noticed you did not replicate the training/testing procedure as it was done in the main paper.
You need to use zero padding and horizontal flip . you can use this [script](https://github.com/Pastromhaug/caffe-stochastic-depth/blob/master/examples/cifar10/preprocessing.py) for that. 

Also please note that for AdaDela you need to start with the learning rate of 0.1 and then divide that by 10 for each step. meaning, for the first 50K iterations you should be using 0.1, from 50-95 I guess, 0.01, and so on . you can see the actual learning rates in the log files such as this [one ](https://raw.githubusercontent.com/Coderx7/SimpleNet/master/SimpNet_V1/Logs/CIfar10/zeropad/caffe_95.32.log) which look like the following (notice the` Iteration xxx, lr = yyy` :   

```
I0310 23:44:29.277779 16900 sgd_solver.cpp:106] Iteration 0, lr = 0.1
I0310 23:44:46.959233 16900 solver.cpp:228] Iteration 100, loss = 1.82308
I0310 23:44:46.959718 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.31
I0310 23:44:46.959718 16900 solver.cpp:244]     Train net output #1: loss = 1.82308 (* 1 = 1.82308 loss)
I0310 23:44:46.959718 16900 sgd_solver.cpp:106] Iteration 100, lr = 0.1
I0310 23:45:04.605721 16900 solver.cpp:228] Iteration 200, loss = 1.96756
I0310 23:45:04.606207 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.31
I0310 23:45:04.606207 16900 solver.cpp:244]     Train net output #1: loss = 1.96756 (* 1 = 1.96756 loss)
I0310 23:45:04.606207 16900 sgd_solver.cpp:106] Iteration 200, lr = 0.1
I0310 23:45:22.255661 16900 solver.cpp:228] Iteration 300, loss = 1.65577
I0310 23:45:22.255661 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.41
I0310 23:45:22.255661 16900 solver.cpp:244]     Train net output #1: loss = 1.65577 (* 1 = 1.65577 loss)
I0310 23:45:22.255661 16900 sgd_solver.cpp:106] Iteration 300, lr = 0.1


...

I0311 02:26:32.068109 16900 sgd_solver.cpp:106] Iteration 50500, lr = 0.01
I0311 02:26:49.988926 16900 solver.cpp:228] Iteration 50600, loss = 0.250841
I0311 02:26:49.989426 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.92
I0311 02:26:49.989426 16900 solver.cpp:244]     Train net output #1: loss = 0.250842 (* 1 = 0.250842 loss)
I0311 02:26:49.989426 16900 sgd_solver.cpp:106] Iteration 50600, lr = 0.01
I0311 02:27:07.767356 16900 solver.cpp:228] Iteration 50700, loss = 0.234636
I0311 02:27:07.767356 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.94
I0311 02:27:07.767356 16900 solver.cpp:244]     Train net output #1: loss = 0.234636 (* 1 = 0.234636 loss)
I0311 02:27:07.767356 16900 sgd_solver.cpp:106] Iteration 50700, lr = 0.01
I0311 02:27:25.576040 16900 solver.cpp:228] Iteration 50800, loss = 0.227445
I0311 02:27:25.576040 16900 solver.cpp:244]     Train net output #0: accuracy_training = 0.93
I0311 02:27:25.576040 16900 solver.cpp:244]     Train net output #1: loss = 0.227445 (* 1 = 0.227445 loss)
I0311 02:27:25.576040 16900 sgd_solver.cpp:106] Iteration 50800, lr = 0.01
...
and so on
```

Also make sure you apply the weight decay (which is 0.005 by the way and has a noticeable impact on the accuracy you get) properly to all layers, and use the same BN settings as well. 
There is nothing fancy about the architecture, when you configure it properly you should be able to achieve the mentioned accuracy in the paper. 

By the way nice work :)
Good Luck

**Comment by Coderx7** [comment]
By the way, the [pytorch implementation](https://github.com/Coderx7/SimpleNet_Pytorch) is out, looking at this implementation may help you better port this to other frameworks/libraries.
